test/internal/SwappableEventBaseTest.cpp

```diff
@@ -0,0 +1,188 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include <folly/ExceptionWrapper.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "src/internal/SwappableEventBase.h"
+
+using SwappableEventBase = rsocket::SwappableEventBase;
+
+namespace {
+
+// helpers for defining new eventbases/"did this callback run" trackers
+#define EB(name) auto& name = get_event_base()
+#define MAKE_DID_EXEC(name) \
+  auto name = make_did_exec_tracker_impl(__LINE__, __FILE__, #name)
+
+struct DidExecTracker {
+  const int line;
+  const std::string file;
+  const std::string name;
+  DidExecTracker(int line, std::string file, std::string name)
+    : line(line), file(file), name(name) {}
+  MOCK_METHOD0(mark, void());
+};
+
+struct DETMarkedOnce : public ::testing::CardinalityInterface {
+  explicit DETMarkedOnce(DidExecTracker const& det) : det(det) {}
+  DidExecTracker const& det;
+
+  virtual int ConservativeLowerBound() const override { return 1; }
```

Comment:
nit: in fb the coding style recommends avoiding repeating virtual keyword when override is present.